go/plugins/vertexai/vertexai.go

```diff
@@ -213,13 +211,77 @@ func (g *generator) generate(ctx context.Context, input *ai.GenerateRequest, cb
 	// TODO: gm.ToolConfig?
 
 	// Send out the actual request.
-	resp, err := cs.SendMessage(ctx, parts...)
-	if err != nil {
-		return nil, err
+	if cb == nil {
+		resp, err := cs.SendMessage(ctx, parts...)
+		if err != nil {
+			return nil, err
+		}
+
+		r := translateResponse(resp)
+		r.Request = input
+		return r, nil
 	}
 
-	r := translateResponse(resp)
-	r.Request = input
+	// Streaming version.
+	iter := cs.SendMessageStream(ctx, parts...)
+	r := &ai.GenerateResponse{Request: input, Candidates: make([]*ai.Candidate, input.Candidates)}
+	for {
+		chunk, err := iter.Next()
+		if err != nil {
+			if err.Error() == "no more items in iterator" {
```

Comment:
My understanding is that you can write

    if err == iterator.Done

where iterator is google.golang.org/api/iterator/.

https://github.com/googleapis/google-cloud-go/wiki/Iterator-Guidelines

Reply:
Correct.

Reply:
I tried that. I couldn't get it to work. But of course, now it works. Maybe that was fixed somewhere between 0.7.1 and 0.10.0.
And of course, it isn't documented how you should do it.